File: kickstart/2020/A/plates/solution.py
t = int(input())
for test in range(1, t + 1):
    n, k, p = map(int, input().split())
    cums = []
    for _ in range(n):
        beauty = list(map(int, input().split()))
        cum = [0] * (k + 1)
        for i in range(k):
            cum[i + 1] = cum[i] + beauty[i]
        cums.append(cum)

    dp = [[0] * (p + 1) for _ in range(n + 1)]
    for i in range(1, n + 1):
        for j in range(p + 1):
            for x in range(min(j, k) + 1):
                dp[i][j] = max(dp[i][j], dp[i - 1][j - x] + cums[i - 1][x])

    print("CASE #{}: {}".format(test, dp[n][p]))

File: kickstart/2020/A/plates/solution.cpp
#include <iostream>
using namespace std;

int main() {
    int t, n, k, p, beauty;
    cin >> t;
    for (int c = 1; c <= t; ++c) {
        cin >> n >> k >> p;
        int sum[n][k + 1];
        for (int i = 0; i < n; ++i) {
            sum[i][0] = 0;
            for (int j = 0; j < k; ++j) {
                cin >> beauty;
                sum[i][j + 1] = sum[i][j] + beauty;
            }
        }
        int dp[n + 1][p + 1];
        fill(*dp, *dp + (n + 1) * (p + 1), 0);
        for (int i = 1; i < n + 1; ++i) {
            for (int j = 0; j < p + 1; ++j) {
                for (int x = 0; x < min(j,k) + 1; ++x) {
                    dp[i][j] = max(dp[i][j], dp[i - 1][j - x] + sum[i - 1][x]);
                }
            }
        }
        cout << "Case #" << c << ": " << dp[n][p] << endl;
    }
}